File: spring22/T2/NLP/ruche_examples-master/examples_formation/hello_mpi.c
# include <stdlib.h>
# include <stdio.h>
# include <time.h>

# include "mpi.h"

int main ( int argc, char *argv[] );

/******************************************************************************/

int main ( int argc, char *argv[] )
{
  int id;
  int ierr;
  int p;
/*
  Initialize MPI.
*/
  ierr = MPI_Init(&argc,&argv);
/*
  Get the number of processes.
*/
  ierr = MPI_Comm_size(MPI_COMM_WORLD, &p);
/*
  Get the individual process ID.
*/
  ierr = MPI_Comm_rank(MPI_COMM_WORLD, &id);
/*
  Process 0 prints an introductory message.
*/
  if (id==0) 
  {
    printf("The number of processes is %d.\n", p);
  }
/*
  Every process prints a hello.
*/
  printf("Hello, world! from process %d\n", id);
/*
  Terminate MPI.
*/
  ierr = MPI_Finalize();

  return 0;
}

File: spring22/T2/NLP/ruche_examples-master/examples_formation/hello_mpi.f90
program main
  use mpi

  integer::error,id,p
!
!  Initialize MPI.
!
  call MPI_Init(error )
!
!  Get the number of processes.
!
  call MPI_Comm_size(MPI_COMM_WORLD,p,error)
!
!  Get the individual process ID.
!
  call MPI_Comm_rank(MPI_COMM_WORLD,id,error)
!
!  Print a message.
!
  if (id==0) then
    write(*,'(a,i1)') 'The number of MPI processes is ',p
  end if
!
!  Every MPI process will print this message.
!
  write ( *, '(a,i1)' ) 'Hello, world! from process ',id
!
!  Shut down MPI.
!
  call MPI_Finalize(error)
end program main
